src/lib.rs

```diff
@@ -1215,10 +1215,11 @@ impl<const N: usize, T> CircularBuffer<N, T> {
     pub fn swap(&mut self, i: usize, j: usize) {
         assert!(i < self.size, "i index out-of-bounds");
         assert!(j < self.size, "j index out-of-bounds");
-        // SAFETY: these are valid pointers
-        unsafe {
-            ptr::swap(self.get_maybe_uninit(i) as *const MaybeUninit<T> as *mut MaybeUninit<T>,
-                      self.get_maybe_uninit(j) as *const MaybeUninit<T> as *mut MaybeUninit<T>);
+        if i != j {
+            let i = add_mod(self.start, i, N);
+            let j = add_mod(self.start, j, N);
+            // SAFETY: these are valid pointers
+            unsafe { ptr::swap_nonoverlapping(&mut self.items[i], &mut self.items[j], 1) };
```

Comment:
It's possible that `i == j`, in which case using `swap_nonoverlapping` would be UB. I would either stick to using `swap`, or add a check to return early if `i == j`.

Reply:
I test against that further up in line 1218. `i` and `j` are both `< N` (checked in the assertions), so the results after applying `add_mod` cannot be the same, either.

Reply:
Oh, right, sorry, somehow I missed that. Looks good to go then!